File: Cloude/src/main/clojure/cloud/cloud/main_clojure.clj
(ns cloud.cloud.main-clojure
  (:require [cloud.cloud.client :as cloud]))

(cloud/defcloudfn square [x]
  (* x x))

(cloud/defcloudfn even-remote? [x]
  (zero? (mod x 2)))

(cloud/defcloudfn times10 [x]
  (* x 10))

(cloud/defcloudfn sum2 [a b]
  (+ a b))

(defn -main [& _]
  (let [legacy-client (cloud/connect "http://localhost:8085")
        legacy-result (cloud/execute-annotated legacy-client #'square [2 3 4 5])
        stream-result (-> (cloud/stream-connect "http://localhost:8085")
                          (cloud/stream [1 2 3 4 5 6])
                          (cloud/filter-annotated #'even-remote?)
                          (cloud/map-annotated #'times10)
                          (cloud/reduce-annotated #'sum2)
                          (cloud/execute-stream))]
    (println "[clojure][legacy] squares:" legacy-result)
    (println "[clojure][pipeline] even*10 sum:" stream-result)))

File: Cloude/src/main/java/cloud/cloud/MainClojure.java
package cloud.cloud;

import java.io.IOException;
import java.util.List;

public class MainClojure {
    public static void main(String[] args) throws IOException, InterruptedException {
        ClojureCloud clojureCloud = ClojureCloud.connect("http://localhost:8085");

        List<Integer> legacyResult = clojureCloud.execute("(fn [x] (* x x))", new int[]{2, 3, 4, 5});
        System.out.println("[clojure][legacy] squares: " + legacyResult);

        List<Object> streamResult = ClojureCloudStream.connect("http://localhost:8085")
                .stream(new int[]{1, 2, 3, 4, 5, 6})
                .filter("(fn [x] (zero? (mod x 2)))")
                .map("(fn [x] (* x 10))")
                .reduce("(fn [a b] (+ a b))")
                .execute();

        System.out.println("[clojure][pipeline] even*10 sum: " + streamResult);
    }
}
